src/allocation/direct-grants/template.sol

```diff
@@ -64,11 +74,26 @@ contract DirectGrants is IAllocationStrategy, Initializable {
         // decode data to get list of 
         //  - identityId
         //  - index of application (to know which milestone)
+        address[] memory identityIds = abi.decode(_data, (address[]));
+
+        for (uint i = 0; i < identityIds.length; i++) {
+            // get application from applications mapping
+            // check if application milestone is accepted (lookup applications mapping)
+            // update application to status to ALLOCATED and make payment
+            // emit event
+        }
 
         // check if application milestone is accepted (lookup applications mapping)
         // update application to status to ALLOCATED and set allocatedAmount
         // emit event
-        // return amount;
+        return 1;
+    }
+
+    function generatePayouts() external view override returns (bytes memory) {
```

Comment:
What happened to the text we had here ?

```
        // TODO: WHAT TO DO HERE? HOW DO WE UPDATE STATUS OF MILESTONE TO PAID ?
        // MIGHT HAVE TO DISTRIBUTION STRATEGY TO DO THIS
        // SHOULD THIS HAVE ARGUMENT TO PASS IN LIST OF MILESTONES TO PAY OUT ?

        // There are 2 ways to do this
        //  - Anytime this is invoked, it generates a list of all the ALLOCATED application milestone
        //    even if it's paid out. It would be upto the distribution strategy to check if it's already paid out. 
        //    Downside: would result in very custom distribution strategy. CANNOT USE EXISTING ONES (as they don't track)   

        //  - Another option is having a callback function which the distribution strategy would invoke after paying out
        //    so that it status here can be marked as PAID. This again results in custom distribution strategy cause they have to 
        //    invoke this callback function. Maybe we add this callback to the IAllocationStrategy interface and 
        //    expect IDistribution.activateDistribution to invoke this callback function ALWAYS.
        ```